File: backend/app/schemas/statistics.py
"""统计相关 schemas。"""

from pydantic import BaseModel


class ProjectStatsItem(BaseModel):
    """项目统计项。"""

    project_id: int
    project_name: str
    total: int
    passed: int


class OverviewStats(BaseModel):
    """整体概览统计。"""

    total_persons: int
    today_new: int
    week_new: int
    month_new: int
    status_breakdown: dict[str, int]
    conversion_rate: float
    pending_reminders: int
    project_stats: list[ProjectStatsItem]

File: frontend/src/types/statistics.ts
export interface ProjectStatsItem {
  project_id: number
  project_name: string
  total: number
  passed: number
}

export interface OverviewStats {
  total_persons: number
  today_new: number
  week_new: number
  month_new: number
  status_breakdown: Record<string, number>
  conversion_rate: number
  pending_reminders: number
  project_stats: ProjectStatsItem[]
}

export interface FunnelItem {
  status: string
  count: number
  percentage: number
}

export interface DailyStats {
  date: string
  new_count: number
  passed_count: number
  failed_count: number
}

export interface FailReasonBreakdown {
  reason: string
  count: number
  percentage: number
}
